### PR TITLE
Sites 269

### PR DESCRIPTION
**Pull request by minorwm**
# READY FOR REVIEW

# Summary
- drush registry-rebuild disables install profile of migrated sites.

# Needed By (Date)
- Yesterday

# Criticality
- How critical is this PR on a 1-10 scale? 5/10
- It affects `migration-sync-only-playbook.yml` and `full-migration-playbook.yml`

# Steps to Test
1. Run a site migration test using the spur-b dept. 

2. `drush @acsf.dev.cardinald7.suprib sqlq 'select name, status from system where name="stanford_dept"'` should return 1.

3. `drush @acsf.dev.cardinald7.suprib rr`

4. `drush @acsf.dev.cardinald7.suprib sqlq 'select name, status from system where name="stanford_dept"'` should STILL return 1.

# Affected Projects or Products
- Group/dept site migrations

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/SITES-269
- Other PRs: N/A
- Any other contextual information that might be helpful: Here was the lead I went off of ( https://www.drupal.org/project/drupal/issues/1170362#comment-12823812 ). I found that adding the step to enable the stanford_dept install profile fixes this issue. I tried adding this to the "Set site up as a Department site" task in the post-db-restore role, but it didn't work. I moved it further down and it did. There's probably something in-between the post-db-restore role and the acsf-optimizations role that's causing this.
- Anyone who should be notified? ( @jbickar  )

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by jbickar** [comment]
@sherakama, how do you feel about enabling the install profile? This does resolve the issue, but it also re-runs the install tasks, and generates some PDO exceptions:

```
2018-11-28 10:11:47,990 p=4848 u=jbickar |  failed: [supri-b] (item=-y en stanford_dept) => {"changed": true, "cmd": "drush      @acsf.test.cardinald7.suprib -y en stanford_dept", "delta": "0:00:03.978300", "end": "2018-11-28 10:11:47.972099", "item": "     -y en stanford_dept", "msg": "non-zero return code", "rc": 1, "start": "2018-11-28 10:11:43.993799", "stderr": "", "stderr_l     ines": [], "stdout": "The following extensions will be enabled: stanford_dept, comment\nDo you really want to continue? (y/n     ): y\nPDOException: SQLSTATE[23000]: Integrity constraint violation: 1062      [error]\nDuplicate entry 'stanford_wilbur-sys     tem-main' for key 'tmd' in\n/mnt/www/html/cardinald702test/docroot/includes/database/database.inc:2227\nStack trace:\n#0\n/m     nt/www/html/cardinald702test/docroot/includes/database/database.inc(2227):\nPDOStatement->execute(Array)\n#1\n/mnt/www/html/     cardinald702test/docroot/includes/database/database.inc(697):\nDatabaseStatementBase->execute(Array, Array)\n#2\n/mnt/www/ht     ml/cardinald702test/docroot/includes/database/mysql/query.inc(36):\nDatabaseConnection->query('INSERT INTO {bl...', Array, A     rray)\n#3\n/mnt/www/html/cardinald702test/docroot/profiles/standard/standard.install(200):\nInsertQuery_mysql->execute()\n#4     \n/mnt/www/html/cardinald702test/docroot/profiles/stanford/stanford.install(11):\nstandard_install()\n#5\n/mnt/www/html/card     inald702test/docroot/profiles/stanford_dept/stanford_dept.install(13):\nstanford_install()\n#6 /mnt/www/html/cardinald702tes     t/docroot/includes/module.inc(934):\nstanford_dept_install()\n#7 /mnt/www/html/cardinald702test/docroot/includes/module.inc(     498):\nmodule_invoke('stanford_dept', 'install')\n#8\n/usr/local/drush7/vendor/drush/drush/commands/core/drupal/environment_     7.inc(143):\nmodule_enable(Array)\n#9\n/usr/local/drush7/vendor/drush/drush/commands/pm/pm.drush.inc(1120):\ndrush_module_en     able(Array)\n#10 /usr/local/drush7/vendor/drush/drush/includes/command.inc(359):\ndrush_pm_enable('stanford_dept')\n#11 /usr     /local/drush7/vendor/drush/drush/includes/command.inc(210):\n_drush_invoke_hooks(Array, Array)\n#12 /usr/local/drush7/vendor     /drush/drush/includes/command.inc(178):\ndrush_command('stanford_dept')\n#13\n/usr/local/drush7/vendor/drush/drush/lib/Drush     /Boot/BaseBoot.php(62):\ndrush_dispatch(Array)\n#14 /usr/local/drush7/vendor/drush/drush/drush.php(70):\nDrush\\Boot\\BaseBo     ot->bootstrap_and_dispatch()\n#15 /usr/local/drush7/vendor/drush/drush/drush.php(11): drush_main()\n#16 {main}", "stdout_lin     es": ["The following extensions will be enabled: stanford_dept, comment", "Do you really want to continue? (y/n): y", "PDOEx     ception: SQLSTATE[23000]: Integrity constraint violation: 1062      [error]", "Duplicate entry 'stanford_wilbur-system-main'      for key 'tmd' in", "/mnt/www/html/cardinald702test/docroot/includes/database/database.inc:2227", "Stack trace:", "#0", "/mn     t/www/html/cardinald702test/docroot/includes/database/database.inc(2227):", "PDOStatement->execute(Array)", "#1", "/mnt/www/     html/cardinald702test/docroot/includes/database/database.inc(697):", "DatabaseStatementBase->execute(Array, Array)", "#2", "     /mnt/www/html/cardinald702test/docroot/includes/database/mysql/query.inc(36):", "DatabaseConnection->query('INSERT INTO {bl.     ..', Array, Array)", "#3", "/mnt/www/html/cardinald702test/docroot/profiles/standard/standard.install(200):", "InsertQuery_m     ysql->execute()", "#4", "/mnt/www/html/cardinald702test/docroot/profiles/stanford/stanford.install(11):", "standard_install(     )", "#5", "/mnt/www/html/cardinald702test/docroot/profiles/stanford_dept/stanford_dept.install(13):", "stanford_install()",      "#6 /mnt/www/html/cardinald702test/docroot/includes/module.inc(934):", "stanford_dept_install()", "#7 /mnt/www/html/cardinal     d702test/docroot/includes/module.inc(498):", "module_invoke('stanford_dept', 'install')", "#8", "/usr/local/drush7/vendor/dr     ush/drush/commands/core/drupal/environment_7.inc(143):", "module_enable(Array)", "#9", "/usr/local/drush7/vendor/drush/drush     /commands/pm/pm.drush.inc(1120):", "drush_module_enable(Array)", "#10 /usr/local/drush7/vendor/drush/drush/includes/command.     inc(359):", "drush_pm_enable('stanford_dept')", "#11 /usr/local/drush7/vendor/drush/drush/includes/command.inc(210):", "_dru     sh_invoke_hooks(Array, Array)", "#12 /usr/local/drush7/vendor/drush/drush/includes/command.inc(178):", "drush_command('stanf     ord_dept')", "#13", "/usr/local/drush7/vendor/drush/drush/lib/Drush/Boot/BaseBoot.php(62):", "drush_dispatch(Array)", "#14...
```

File: roles/acsf-optimizations/tasks/main.yml [hunk 1]
```diff
@@ -38,6 +38,10 @@
 - name: Disable error reporting
   shell: "{{ drush_alias }} -y --exact vset error_level 0"
 
+#- name: Enable stanford_dept again to fix when drush rr disables it
```
Comment (jbickar) [inline roles/acsf-optimizations/tasks/main.yml]: I'll remove these commented-out lines if we decide to merge this.

File: roles/post-db-restore/tasks/main.yml [hunk 2]
```diff
@@ -62,11 +62,13 @@
   with_items:
     - "-y vset install_profile stanford_dept"
     - "cc all"
-    - "sqlq 'update system set status=\"1\" where name=\"stanford_dept\"'"
+#    - "sqlq 'update system set status=\"1\" where name=\"stanford_dept\"'"
+    - "-y en stanford_dept"
```
Comment (jbickar) [inline roles/post-db-restore/tasks/main.yml]: @minorwm, note that I moved this to here...

File: roles/post-db-restore/tasks/main.yml [hunk 3]
```diff
@@ -62,11 +62,13 @@
...
     - "-y en acsf"
     - "rr"
   when: dept_site == "TRUE"
   notify: Clear site cache
+  ignore_errors: "{% if ignore_updb_errors == 'TRUE' %}yes{% endif %}"
```
Comment (jbickar) [inline roles/post-db-restore/tasks/main.yml]: ...and added some error handling here.

**Comment by jbickar** [comment]
@sherakama, thoughts on this approach?

**Comment by sherakama** [comment]
If we don't need to install the profile again instead of running `drush en stanford_profile` do a mysql query that sets the enabled status of the module instead. `drush sqlq` is your friend here.

Also, does this resolve the issue for sites that I manually run `drush rr` on later? If not, I would like to see that resolved as well as that seems like the deeper issue.

File: roles/post-db-restore/tasks/main.yml [hunk 4]
```diff
@@ -62,11 +62,13 @@
   with_items:
     - "-y vset install_profile stanford_dept"
     - "cc all"
-    - "sqlq 'update system set status=\"1\" where name=\"stanford_dept\"'"
+#    - "sqlq 'update system set status=\"1\" where name=\"stanford_dept\"'"
```
Comment (jbickar) [inline roles/post-db-restore/tasks/main.yml]: @sherakama, yeah the problem is that `drush sqlq 'update system set status="1" where name="stanford_dept"'` doesn't work.

Bill's approach _does_ resolve the issue with running `drush rr` later. I'm just not sure about the ramifications of re-running the install profile. I'm trying to figure out what `drush en` is doing that the SQL query is not.

Comment (sherakama) [reply roles/post-db-restore/tasks/main.yml]: Running the installation profile after the database restore sounds like a bad idea as the profiles execute a lot of configuration code and could enable dependencies that you do not want. I would advise against running the install.

Comment (sherakama) [reply roles/post-db-restore/tasks/main.yml]: https://api.drupal.org/api/drupal/includes%21module.inc/function/module_enable/7.x

I would start here

Comment (jbickar) [reply roles/post-db-restore/tasks/main.yml]: I wonder if registry rebuild disables the profile because some dependencies aren't enabled.

**Comment by jbickar** [comment]
See #79

**Pull request closed without merging**